components/brave_wallet/browser/internal/hd_key_common.cc

```diff
@@ -23,6 +23,15 @@ DerivationIndex DerivationIndex::Hardened(uint32_t index) {
   return DerivationIndex(index, true);
 }
 
+// static
+DerivationIndex DerivationIndex::FromRawValueForTesting(uint32_t index) {
```

Comment:
IIRC, we don't want to allow ED25519 to use non-hardened options as it will allow for that attack I investigated previously. Can you clarify what's this for then since I assume it's not for that given we just recently looked at it?

Reply:
It is not about ED25519 itself, but the way private keys / scalars are hierarchically generated. For SLIP10 it is not possible to do normal derivations, but for SLIP23 it is.